devito/grid.py

```diff
@@ -58,8 +65,25 @@ def __init__(self, shape, extent=None, origin=None, dimensions=None,
         # TODO: Raise proper exceptions and logging
         assert(self.dim == len(self.origin) == len(self.extent) == len(self.spacing))
 
-        # TODO: Create Dimensions locally instead of using global ones
-        self.dimensions = (dimensions or (x, y, z))[:self.dim]
+        if dimensions is None:
+            # Create the spatial dimensions and constant spacing symbols
+            assert(self.dim <= 3)
+            dim_names = self._default_dimensions[:self.dim]
+            dim_spacing = tuple(Constant(name='h_%s' % name, value=val)
+                                for name, val in zip(dim_names, self.spacing))
+            self.dimensions = tuple(SpaceDimension(name=name, spacing=spc)
+                                    for name, spc in zip(dim_names, dim_spacing))
+        else:
+            self.dimensions = dimensions
+
+        # Store or create default symbols for time and stepping dimensions
+        if time_dimension is None:
+            self.time_dim = TimeDimension('time', spacing=Constant(name='dt'))
```

Comment:
Yeah, a free function sounds sensible. The globally available dummy grid, however, won;t work, since the grid now adds default spacing values to the `Constant` spacing symbols, so it's not generic enough.